test/DebugInfo/Inputs/type-reconstr-names.txt

```diff
@@ -2,3 +2,4 @@ $S4blah4mainyyF ---> () -> ()
 $SytD ---> ()
 $Ss5Int32VD ---> Int32
 $S4blah4mainyyF8PatatinoL_VMa ---> Can't resolve type of $S4blah4mainyyF8PatatinoL_VMa
+$Ss10CollectionP7Element ---> Collection
```

Comment:
I don't think this is a valid mangling. Where did it come from? Note that it has the identifier 'Element'. Here's the mangling tree:

```
kind=Global
  kind=Protocol
    kind=Module, text="Swift"
    kind=Identifier, text="Collection"
  kind=Identifier, text="Element"
```

This doesn't look right.

Reply:
It's coming from `swift-ide-test -reconstruct-type` which calls the mangler on `extension Collection`. It could possibly be a mangler bug (or a bug in `swift-ide-test`), I'll take a look tomorrow, thanks for noticing.

Regardless, I think we shouldn't crash if we're passing an invalid mangled name. If we don't want to just ignore the identifier here, maybe we should have a previous validation step?

cc: @eeckstein

Reply:
I think the node visitor needs to be refactored so that each step can fail and we gracefully unwind and return a null Type.

Reply:
@dcci Can you figure out why that mangled name was being emitted and from where please? Bugs in how we mangle types for debug info are just as bad as bugs in type reconstruction or the demangler.